src/m365/aad/commands/user/user-remove.spec.ts

```diff
@@ -0,0 +1,208 @@
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogToStderrSpy = sinon.spy(logger, 'logToStderr');
+    sinon.stub(Cli, 'prompt').callsFake(async (options: any) => {
+      promptOptions = options;
+      return { continue: false };
+    });
+    promptOptions = undefined;
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.delete,
+      Cli.prompt
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.USER_REMOVE);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if id is not a valid guid.', async () => {
+    const actual = await command.validate({
+      options: {
+        id: 'Invalid GUID'
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified (id)', async () => {
+    const actual = await command.validate({ options: { id: validId } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified (name)', async () => {
+    const actual = await command.validate({ options: { userName: validUsername } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('prompts before removing the specified user when confirm option not passed', async () => {
+    await command.action(logger, {
+      options: {
+        id: validId
+      }
+    });
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing the specified user when confirm option not passed and prompt not confirmed', async () => {
+    const postSpy = sinon.spy(request, 'delete');
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: false }
+    ));
+    await command.action(logger, {
+      options: {
+        id: validId
+      }
+    });
+    assert(postSpy.notCalled);
+  });
+
+  it('removes the specified user by id', async () => {
+    sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${validId}`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: true }
+    ));
+    await command.action(logger, {
+      options: {
+        debug: true,
+        id: validId
+      }
+    });
+    assert(loggerLogToStderrSpy.called);
```

Comment:
Right now we're only checking whether we are logging a verbose message.
Instead, we should check whether we actually fired the `delete` request.